### PR TITLE
[Feature]   create new functions such as current_timestamp, current_date

### PR DESCRIPTION
**Pull request by shshenhua**
…amp funcation

like spark and presto

## What type of PR is this：
- [x] Feature

## Which issues of this PR fixes ：

Fixes 
https://github.com/StarRocks/starrocks/issues/15382 
https://github.com/StarRocks/starrocks/issues/15285

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 7 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[4548, 4549, 4552]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|6|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]
better add UT

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 3 / 4 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|4|75.00%|[4552]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|7|85.71%|[4605]|

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|6|7|85.71%|[4605]|

**Review by Astralidea — DISMISSED** [review]

**Comment by shshenhua** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -1777,6 +1778,14 @@ informationFunctionExpression
...
+    : name = CURRENT_DATE ('(' ')')?
+    | name = CURRENT_TIME ('(' ')')?
+    | name = CURRENT_TIMESTAMP
+    | name = LOCALTIME ('(' ')')?
+    | name = LOCALTIMESTAMP ('(' ')')?
+    ;
+
 specialFunctionExpression
     : CHAR '(' expression ')'
     | CURRENT_TIMESTAMP '(' ')'
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1791]: Can unify CURRENT_TIMESTAMP and specialDateTimeExpression together

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**